cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -463,8 +463,30 @@ func ValidateAPIEndpoint(c *kubeadm.APIEndpoint, fldPath *field.Path) field.Erro
 	return allErrs
 }
 
-// ValidateIgnorePreflightErrors validates duplicates in ignore-preflight-errors flag.
-func ValidateIgnorePreflightErrors(ignorePreflightErrors []string) (sets.String, error) {
+// ValidateIgnorePreflightErrors validates duplicates in:
+// - ignore-preflight-errors flag and
+// - ignorePreflightErrors field in {Init,Join}Configuration files.
+func ValidateIgnorePreflightErrors(ignorePreflightErrorsFromCLI, ignorePreflightErrorsFromConfigFile []string) (sets.String, error) {
+	ignoreErrorsFromConfigFile, err := validateIgnorePreflightErrorsFromConfigFile(ignorePreflightErrorsFromConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	ignoreErrorsFromCLI, err := validateIgnorePreflightErrorsFromCLI(ignorePreflightErrorsFromCLI)
+	if err != nil {
+		return nil, err
+	}
+	if ignoreErrorsFromCLI.Has("all") {
+		// By construction in validateIgnorePreflightErrorsFromCLI, if "all" is
+		// in the set from CLI, then it is the ONLY value it contains, hence we
+		// return ONLY this set.
+		return ignoreErrorsFromCLI, nil
```

Comment:
If 'all' is set in CLI and config has something else in addition, it is an error situation. 
```go
func ValidateIgnorePreflightErrors(ignorePreflightErrorsFromCLI, ignorePreflightErrorsFromConfigFile []string) (sets.String, error) {
	ignoreErrors := sets.NewString()
	allErrs := field.ErrorList{}

	for _, item := range ignorePreflightErrorsFromConfigFile {
		ignoreErrors.Insert(strings.ToLower(item)) // parameters are case insensitive
	}

	if ignoreErrors.Has("all") {
		// "all" is forbidden in config files. Administrators should use an
		// explicit list of errors they want to ignore, as it can be risky to
		// mask all errors in such a way. Hence, we return an error:
		allErrs = append(allErrs, field.Invalid(field.NewPath("ignorePreflightErrors"), "all", "'all' cannot be used in configuration file"))
        }

	for _, item = range ignorePreflightErrorsFromCLI {
		ignoreErrors.Insert(strings.ToLower(item)) // parameters are case insensitive
	}

	if ignoreErrors.Has("all") && ignoreErrors.Len() > 1 {
		allErrs = append(allErrs, field.Invalid(field.NewPath("ignore-preflight-errors"), strings.Join(ignoreErrors.List(), ","), "don't specify individual checks if 'all' is used"))
	}

	return ignoreErrors, allErrs.ToAggregate()
}
```